### PR TITLE
Local aisle

### PR DESCRIPTION
**Pull request by Pike**
Getting my local changes to l20n.js public, which I'm using for my aisle tooling, aka, moz.l20n.

### CODE REVIEW

File: src/lib/format/l20n/ast/parser.js [hunk 1]
```diff
@@ -483,7 +484,11 @@ export default {
 
     this._index = nextEntry;
 
-    return new AST.JunkEntry(
+    var junk = new AST.JunkEntry(
```
Comment (stasm) [inline src/lib/format/l20n/ast/parser.js]: Please use `const`.

File: tools/lib/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ require('colors');
 require('../../node_modules/babel-core/register');
 var PropertiesParser =
   require('../../src/lib/format/properties/parser');
-var L20nParser = require('../../src/lib/format/l20n/entries/parser');
+var L20nParser = require('../../src/lib/format/l20n/ast/parser');
```
Comment (stasm) [inline tools/lib/index.js]: I'll fix that in a different commit.

File: src/lib/format/l20n/ast/parser.js [hunk 3]
```diff
@@ -460,7 +460,8 @@ export default {
 
     let msg = message + ' at pos ' + pos + ': `' + context + '`';
 
-    const err = new L10nError(msg);
+    const err = new L10nError(message);
+    err.context = context
```
Comment (zbraniecki) [inline src/lib/format/l20n/ast/parser.js]: I fixed it in my previous commit.

**Comment by Pike** [comment]
Comments addressed and rebased against master now. I'd think this is ready to merge.

**Comment by stasm** [comment]
@Pike: I think we'll need a new PR because this one wants to merge into v3.x.  Or I can just cherry-pick this for you, if that's okay?

**Comment by Pike** [comment]
Right, I'll just open one for master, that's trivial enough.

**Pull request closed without merging**